addons/controllers/suite_test.go

```diff
@@ -331,6 +341,14 @@ var _ = BeforeSuite(func(done Done) {
 	err = clusterbootstrapTemplateWebhook.SetupWebhookWithManager(mgr)
 	Expect(err).ToNot(HaveOccurred())
 
+	// Setup the tkg-system namespace resources.
+	// We do it here because  specs may be executed in parallel and
+	// this is the only way to assure the tkg-system resources are ready before all specs start
+	f, err := os.Open("testdata/test-tkg-system-ns-resources.yaml")
```

Comment:
This will void the ClusterBootstrap and ClusterBootstrapTemplate webhook tests. The webhooks are enabled in `clusterbootstrap_controller_test.go` before creating these resources. Now we have moved these resources out, they will be created before the webhooks are in place, so won't be validated and tested.

Reply:
I ran all tests and they pass but if I understand correctly, this might have invalidated the webhook tests regardless of their result. 
We need to figure out a way to setup the tkg-system resources needed by all tests before we call any tests. 
in ginkgo v3 there is a before all tests option, but we don't have that here. 
The reason I had to do it, was because since the machine_controller tests are in a different spec, they sometimes get executed before the tkg-system resources are completely setup and they fail. and we end up with a flaky test. 

Also without this change, running things like "ginkgo -v -f "ClusterBootstrap is paused" will fail because the tkg-system resources in test-cluster-bootstrap-1.yaml are not created.  It might seem not a big deal, but it means that specs can't be safely run in parallel. 

also can't source test-cluster-boostrap-1.yaml on every spec (when statement) because it complains about defining the resources more than once.

Reply:
Had a discussion with Adolfo offline. We are going to enable the CB and CBT webhooks in suite_test.go before the resources are applied.
With this, it is enforced that all the testcases that creates CB and CBT need to have all the resources required, otherwise will be rejected by the webhook.